internal/webhooks/clusterclass.go

```diff
@@ -538,33 +539,45 @@ func validateMachineHealthCheckClasses(clusterClass *clusterv1.ClusterClass) fie
 
 	// Validate ControlPlane MachineHealthCheck if defined.
 	if clusterClass.Spec.ControlPlane.MachineHealthCheck != nil {
+		controlPlaneMHCPath := field.NewPath("spec", "controlPlane", "machineHealthCheck")
+
+		allErrs = append(allErrs, validateMachineHealthCheckClass(controlPlaneMHCPath, clusterClass.Namespace,
+			clusterClass.Spec.ControlPlane.MachineHealthCheck)...)
+
 		// Ensure ControlPlane does not define a MachineHealthCheck if it does not define MachineInfrastructure.
 		if clusterClass.Spec.ControlPlane.MachineInfrastructure == nil {
 			allErrs = append(allErrs, field.Forbidden(
-				field.NewPath("spec", "controlPlane", "machineHealthCheck"),
+				controlPlaneMHCPath.Child("machineInfrastructure"),
 				"can be set only if spec.controlPlane.machineInfrastructure is set",
 			))
 		}
-		// Ensure ControlPlane MachineHealthCheck defines UnhealthyConditions.
-		if len(clusterClass.Spec.ControlPlane.MachineHealthCheck.UnhealthyConditions) == 0 {
-			allErrs = append(allErrs, field.Forbidden(
-				field.NewPath("spec", "controlPlane", "machineHealthCheck", "unhealthyConditions"),
-				"must be defined and have at least one value",
-			))
-		}
 	}
 
 	// Ensure MachineDeployment MachineHealthChecks define UnhealthyConditions.
 	for i, md := range clusterClass.Spec.Workers.MachineDeployments {
 		if md.MachineHealthCheck == nil {
 			continue
 		}
-		if len(md.MachineHealthCheck.UnhealthyConditions) == 0 {
-			allErrs = append(allErrs, field.Forbidden(
-				field.NewPath("spec", "workers", "machineDeployments", "machineHealthCheck").Index(i).Child("unhealthyConditions"),
-				"must be defined and have at least one value",
-			))
-		}
+		machineDeploymentMHCPath := field.NewPath("spec", "machineDeployments", "machineHealthCheck").Index(i)
+
+		allErrs = append(allErrs, validateMachineHealthCheckClass(machineDeploymentMHCPath, clusterClass.Namespace, md.MachineHealthCheck)...)
 	}
 	return allErrs
 }
+
+// validateMachineHealthCheckClass validates the MachineHealthCheckSpec fields defined in a MachineHealthCheckClass.
+func validateMachineHealthCheckClass(specPath *field.Path, namepace string, m *clusterv1.MachineHealthCheckClass) field.ErrorList {
```

Comment:
```suggestion
func validateMachineHealthCheckClass(fldPath *field.Path, namepace string, m *clusterv1.MachineHealthCheckClass) field.ErrorList {
```
Let's also use fldPath, not sure what specPath is :)